rfcs/OnError.md

```diff
@@ -0,0 +1,79 @@
+RFC: `onError` request property
+-------
+
+See also [SemanticNullability.md](SemanticNullability.md). 
+
+`onError` is a part of the more general semantic nullability solution. But `onError` can be added independently of the more general solution, hence a separate RFC.
+
+## 📜 Introduction
+
+When working on improving GraphQL nullability, it was found that [GraphQL error propagation](https://spec.graphql.org/draft/#sec-Executing-Selection-Sets.Errors-and-Non-Null-Types), by encouraging the use of nullable types for error reasons is creating confusion for frontend developers. 
+
+"Is a field nullable for product reasons or error reasons?" Without more information, it's impossible to tell 🤷‍♂️.
+
+This defeats GraphQL as a type-safe language aiming to improve backend/frontend communication.
+
+This proposal introduces `onError` as a way to disable error propagation so that the schemas can express the [true nullability](https://github.com/graphql/graphql-wg/discussions/1394) of their data.
+
+## 💡 The `onError` request property
+
+It is proposed that clients send a new `onError` property:
+
+```json
+{
+  "query": "{ helloWorld }",
+  "onError": "NULL"
+}
+```
+
+Possible values for the `onError` property are:
+
+* `NULL`: disables error propagation. The executor emits `null` in the `data` part of the response and a GraphQL [execution error](https://spec.graphql.org/draft/#execution-error) in the `errors` part of the response.
+* `PROPAGATE`: propagates the error to the nearest nullable parent field. This is the current behaviour.
+* `HALT`: stops executing the request whenever an error is encountered. This allows services to save resources if clients do not consume partial data.
+
+Servers not supporting `onError` must execute in the `PROPAGATE` mode. 
+
+Servers supporting `onError` must decide on a sensible default. Existing servers will most likely use `PROPAGATE`, new servers will most likely use `NULL`. `HALT` is also possible as a default albeit it's expected to be used less frequently.
```

Comment:
Would it be simpler if the default is always `PROPAGATE`? That automatically makes this compatible with old clients, and removes the need to surface the default in introspection (unless there's another reason to do so that I missed?)